mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/tts/android/AndroidTextToSpeechFactory.kt

```diff
@@ -0,0 +1,71 @@
+package ch.epfl.sdp.mobile.infrastructure.tts.android
+
+import android.content.Context
+import android.os.Bundle
+import android.speech.tts.TextToSpeech as NativeTextToSpeech
+import ch.epfl.sdp.mobile.infrastructure.tts.TextToSpeech
+import ch.epfl.sdp.mobile.infrastructure.tts.TextToSpeechFactory
+import java.lang.IllegalStateException
+import java.util.*
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+import kotlin.properties.Delegates.notNull
+
+
+class AndroidTextToSpeechFactory(
+    private val context: Context,
+    private val volume: VolumeLevel = VolumeLevel.On
+) : TextToSpeechFactory {
+  override suspend fun create(): TextToSpeech = suspendCoroutine { cont ->
+    var speech by notNull<NativeTextToSpeech>()
+    val callback =
+        NativeTextToSpeech.OnInitListener { status ->
+          when (status) {
+            NativeTextToSpeech.SUCCESS -> cont.resume(AndroidTextToSpeech(speech, volume))
+            NativeTextToSpeech.ERROR -> cont.resumeWithException(IllegalStateException())
+          }
+        }
+
+    speech = NativeTextToSpeech(context, callback)
+  }
+}
+
+class AndroidTextToSpeech(
+    private val tts: NativeTextToSpeech,
+    private val volume: VolumeLevel = VolumeLevel.On,
+) : TextToSpeech {
+  init {
+    // Set up properties like voice, languages ...
+    tts.language = Locale.ENGLISH
+    tts.voice = null
+    tts.setSpeechRate(0f)
+    tts.setPitch(0f)
```

Comment:
Question : what are the defaults for these parameters ? Do we really need to change them ?

Also, consider using some dedicated constants to avoid magic values.